src/Scripting/Core/Hosting/CommandLine/CommandLineRunner.cs

```diff
@@ -192,6 +192,25 @@ private int RunScript(ScriptOptions options, SourceText code, ErrorLogger errorL
                 _compiler.ReportErrors(e.Diagnostics, _console.Error, errorLogger);
                 return CommonCompiler.Failed;
             }
+            catch (Exception e)
+            {
+                var exceptionLines = e.ToString().Split(new[] { Environment.NewLine }, StringSplitOptions.None);
+
```

Comment:
Parsing Exception.ToString() is not a good idea. Please use [https://github.com/dotnet/roslyn/blob/eaea480107b80b081f18b3417c38458ccbe622dc/src/Scripting/Core/Hosting/CommandLine/CommandLineRunner.cs#L317](https://github.com/dotnet/roslyn/blob/eaea480107b80b081f18b3417c38458ccbe622dc/src/Scripting/Core/Hosting/CommandLine/CommandLineRunner.cs#L317).

Reply:
Sorry, I don't follow, how will using `DisplayException()` help? I tried that, and it reduces the output to the exception message only. The idea in this PR is to preserve the stack trace in the output but to strip out the lines which refer to csi internals, leaving only the lines which refer to the script.

Reply:
DispalyException calls http://source.roslyn.io/#Microsoft.CodeAnalysis.Scripting/Hosting/ObjectFormatter/CommonObjectFormatter.cs,58. It shouldn't reduce exception to just a message, except for specific load exceptions. Seems like a bug in the filtering logic.

Reply:
OK, but assuming that bug is fixed, it still won't provide the solution, correct? Or does it know about filtering out stack trace lines that are outside the submission?

Reply:
It will provide solution. What do you mean "outside" of submission? You want to display frames from libraries you call. The exception formatter filters out "helper" frames that are not interesting.

Reply:
@tmat I found the problem: the stack frame we are interested in is of the form:

```
   at Submission#0.<<Initialize>>d__0.MoveNext() in c:\code\foo.csx:line 6
```

When [CommonMemberFilter.IsHiddenMember](http://source.roslyn.io/#Microsoft.CodeAnalysis.Scripting/Hosting/CommonMemberFilter.cs,50) is called for this frame, it returns `true`, since `<<Initialize>>d__0` is a generated member name according to [Microsoft.CodeAnalysis.CSharp.Symbols.GeneratedNames.IsGeneratedMemberName](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Symbols/Synthesized/GeneratedNames.cs,25). Hence, this stack frame is not included in the output. However, this is exactly the line we want in the output.

It seem like `IsHiddenMember` needs to return `false` for any method which is part of the submission?

Reply:
Or perhaps it just needs to special case `Submission#0.<<Initialize>>`. The example above is the entry point to the script, but a lambda which is defined in the script suffers from the same problem, since it has a method name of the form `Submission#0.<<Initialize>>b__0_1()`.

Reply:
I see. This will be an issue for other async methods as well (the top-level code is async). We need some more work done there.